api/client/src/testFixtures/java/org/projectnessie/client/auth/oauth2/ResourceOwnerEmulator.java

```diff
@@ -418,9 +417,17 @@ public void close() throws Exception {
     if (replaceSystemOut) {
       System.setOut(standardOut);
     }
-    executor.shutdownNow();
-    consoleIn.close();
-    consoleOut.close();
+    try {
+      // close writer first to signal end of input to reader
+      consoleOut.close();
+      consoleIn.close();
+    } catch (IOException e) {
+      LOGGER.warn("Error closing console streams", e);
+    }
+    executor.shutdown();
+    if (!executor.awaitTermination(5, TimeUnit.SECONDS)) {
+      executor.shutdownNow();
+    }
```

Comment:
we could consider using Guava's `MoreExecutors.shutdownAndAwaitTermination` instead for this logic